### ISSUE TITLE
allow two 0 columns in import.q.sorts

### ISSUE THREAD

**Issue by maxheld83** [aiorazabala/qmethod#304]
@aiorazabala just suggested:

> The import.q.sorts function to allow two central columns with zero. I see that this is not optimal from a research design point of view, but some people might do it too, and it's not strictly incorrect (plus this restriction requires to code some tweaks to convert the CSV files to the format that the function wants!)

**Comment by maxheld83** [comment]
just to be clear, @aiorazabala, because I don't quite understand why people would use two central columns with zero – this is merely for because data entry is easier (b/c the column height is lower), correct?

**Comment by maxheld83** [comment]
three comments:

- I agree that it is not "wrong", and there is no technical reason not to do it.
- However, `import.q.sorts()` currently *does not look at any row* to figure out the rank, but merely infers it from the index (left to right, from negative to positive).
  So we'd need to add a) a convention for a row in which the column value (-2, -1, 0, 0, 1, 2) etc is stored and b) an *alternative* way to read in q sorts in that way, which, I worry, might make things more complex and buggier.
- Unless people *actually* have people Q-sort items in *two* zero columns (which might be questionable methodologically?), I am somewhat philosophically opposed to accommodating non-raw data (which this would be).

So bottom line: I probably won't spend time doing this, but sure, why not.

**Comment by aiorazabala** [comment]
Hummm, I see the logic of the `import.q.sorts()` function.

1. So basically, having two 0 columns wouldn't matter as long as the bottom row indicates which are the column values, Correct?

2. `if (1 == T)` then the problem would occur when someone enters these Q-sorts but excludes the last line indicating column values, Correct?

**Comment by aiorazabala** [comment]
Here's the code snippet I used to fix this issue when using our 'faulty' data.
`mydata` is a raw matrix like the one you read from a Q-sort csv. In this case, the number of columns is 5.
```{R}
  # Merge the 0 columns
  top.part <- matrix(data=NA, nrow=dim(mydata)[1], ncol=dim(mydata)[2]-1)
  top.part[,4] <- mydata[,5]
  colnames(top.part) <- colnames(mydata[,-5])
  mydata <- rbind(top.part, mydata[,-5])
```

This could be either integrated inside the function (e.g. as an optional argument) or simply point at this snippet whenever someone has this issue.
